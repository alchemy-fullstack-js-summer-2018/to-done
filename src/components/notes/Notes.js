import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Note from './Note';

class Notes extends Component {

  static propTypes = {
    notes: 
    PropTypes.array,
    onUpdate: PropTypes.func.isRequired,
    onRemove: PropTypes.func.isRequired
  };

  render() {
    const { notes, onUpdate, onRemove } = this.props;

    return (
      <ul>
        {notes.map(note => (
          <Note
            key={note.key}
            note={note}
            onUpdate={onUpdate}
            onRemove={onRemove}
          />
        ))}
      </ul>
    );
  }
}
export default Notes;