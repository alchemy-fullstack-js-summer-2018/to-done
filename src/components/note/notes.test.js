import React from 'react';
import { shallow } from 'enzyme';
import Notes from './Notes';
import toJSON from 'enzyme-to-json';

const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

Enzyme.configure({ adapter: new Adapter() });

describe('Notes', () => {
  
  it('renders add if no note prop', () => {
    const notes = [
      { name: 'Laundry', content: 'socks, towels', key: 'abc123' },
      { name: 'Groceries', content: 'bread, pasta, cheese', key: 'abc456' },
      { name: 'Pets', content: 'give kisses and treats', key: 'abc789' }
    ];

    const wrapper = shallow(<Notes
      notes={notes}
      onUpdate={jest.fn()}
      onRemove={jest.fn()}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});