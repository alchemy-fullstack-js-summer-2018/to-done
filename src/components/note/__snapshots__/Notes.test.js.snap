// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes renders add if no note prop 1`] = `
<ul>
  <Note
    key="abc123"
    note={
      Object {
        "content": "socks, towels",
        "key": "abc123",
        "name": "Laundry",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Note
    key="abc456"
    note={
      Object {
        "content": "bread, pasta, cheese",
        "key": "abc456",
        "name": "Groceries",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
  <Note
    key="abc789"
    note={
      Object {
        "content": "give kisses and treats",
        "key": "abc789",
        "name": "Pets",
      }
    }
    onRemove={[MockFunction]}
    onUpdate={[MockFunction]}
  />
</ul>
`;
